pkg/status/status.go

```diff
@@ -143,11 +160,17 @@ func (c *Collector) GetStaleProbes() map[string]time.Time {
 }
 
 // spawnProbe starts a goroutine which invokes the probe at the particular interval.
-func (c *Collector) spawnProbe(p *Probe) {
+func (c *Collector) spawnProbe(p *Probe, firstRunCompleted func()) {
 	go func() {
 		for {
 			c.runProbe(p)
 
+			// The first run of the probe has completed.
+			if firstRunCompleted != nil {
```

Comment:
Non-blocking: maybe just a matter of personal preference, but I find that passing `*lock.StoppableWaitGroup` and call `Done()` on it is more readable.
Also, we don't really need the check against nil, as `spawnProbe` is unexported and called only from `NewCollector`, where we control the initialization of `firstRunSwg`

Reply:
Hmm, AFAIK there's no way of correctly using a StoppableWaitGroup here (without the changes from https://github.com/cilium/cilium/pull/35892), because there would be risk that the same probe calls `Done()` multiple times before other probes even complete once. That's the same reason why I had the `!= nil` check, so that it is gets called only once, and then reset (another possibility could be having an explicit variable, which is a bit more verbose).

Reply:
> That's the same reason why I had the != nil check

Oh, my bad, I misread the code. I'm fine with the `nil` check, then :+1: 

> another possibility could be having an explicit variable, which is a bit more verbose

Yeah, I'd prefer that, but as I said just personal preference, no strong opinion. :+1: